test/functional/zmq_test.py

```diff
@@ -6,29 +6,43 @@
 import configparser
 import os
 import struct
+import time
 
 from test_framework.test_framework import BitcoinTestFramework, SkipTest
-from test_framework.util import *
+from test_framework.util import (assert_equal,
+                                 bytes_to_hex_str,
+                                 )
+
+def wait_for_multipart(socket, timeout=60):
+    interval = 0.1
```

Comment:
You can also call socket.SET(zmq.RCVTIMEO, 60000) when creating the socket instead of the NOBLOCK and the loop. It will raise zmq.error.Again when time is over